### PR TITLE
Add modal window treatment to macOS

### PR DESCRIPTION
**Pull request by Gasparoken**
This fix is the first step to solve issue aseprite/aseprite#4561

### CODE REVIEW

File: os/osx/window.mm [hunk 1]
```diff
@@ -120,6 +120,8 @@ - (WindowOSXObjc*)initWithImpl:(os::WindowOSX*)impl
     if (spec->floating()) {
       self.level = NSFloatingWindowLevel;
       self.hidesOnDeactivate = true;
+      if (spec->modal())
+        [self setLevel:NSModalPanelWindowLevel];
     }
```
Comment (dacap) [inline os/osx/window.mm:125]: We're already setting the level property if it's floating, probably we can have a modal / not floating (this is not the case of Aseprite which sets the "floating" property for all non-main windows):
```suggestion
    if (spec->floating()) {
      self.level = NSFloatingWindowLevel;
      self.hidesOnDeactivate = true;
    }
    if (spec->modal())
      self.level = NSModalPanelWindowLevel;
```

**Review by dacap — CHANGES_REQUESTED** [review]

File: os/osx/window.mm [hunk 2]
```diff
@@ -122,6 +122,9 @@ - (WindowOSXObjc*)initWithImpl:(os::WindowOSX*)impl
       self.hidesOnDeactivate = true;
     }
 
+    if (spec->modal())
+      [self setLevel:NSModalPanelWindowLevel];
```
Comment (dacap) [inline os/osx/window.mm:126]:
```suggestion
      self.level = NSModalPanelWindowLevel;
```

**Comment by dacap** [comment]
I'll merge this, it would be nice to block the whole hierarchy of parent's floating windows when a modal is displayed, but at the moment this works better than the current `main` to fix the bug.

**Comment by Gasparoken** [comment]
> I'll merge this, it would be nice to block the whole hierarchy of parent's floating windows when a modal is displayed, but at the moment this works better than the current `main` to fix the bug.

Great! Remember to approve https://github.com/aseprite/aseprite/pull/4835 so that issue https://github.com/aseprite/aseprite/issues/4561 is resolved in Aseprite.

**Pull request merged**